lib/mongoid/document.rb

```diff
@@ -281,20 +296,39 @@ module ClassMethods
       # @param [ Hash ] attrs The hash of attributes to instantiate with.
       # @param [ Integer ] selected_fields The selected fields from the
       #   criteria.
+      # @param [ true | false ] defer_callbacks Flag specifies whether callbacks
+      #   should be run.
       #
       # @return [ Document ] A new document.
-      def instantiate(attrs = nil, selected_fields = nil)
+      def instantiate(attrs = nil, selected_fields = nil, defer_callbacks = false)
```

Comment:
If we agree that this method is part of Mongoid's public API, I would prefer its API to stay the same as before (like with `initialize`) and another method added to deal with the callback behavior.

Reply:
done